File: backend/model/customer.go
package model

type Customer struct {
	ID             int            `json:"id,omitempty"`
	Name           string         `json:"name,omitempty"`
	Country        Country        `json:"country"`
	Company        string         `json:"company,omitempty"`
	Date           string         `json:"date,omitempty"`
	Status         string         `json:"status,omitempty"`
	Verified       bool           `json:"verified,omitempty"`
	Activity       int            `json:"activity,omitempty"`
	Balance        float64        `json:"balance,omitempty"`
	Representative Representative `json:"representative"`
}

File: frontend/src/app/models/customer.ts
import {Representative} from "./representative";
import {Country} from "./country";

export type Customer = {
  id?: number;
  name?: string;
  country?: Country;
  company?: string;
  date?: string;
  status?: string;
  activity?: number;
  representative?: Representative;
}

export type CustomersListResponse = {
  customers: Customer[];
  totalRecords: number;
}
